### PR TITLE
Git backend

### PR DESCRIPTION
**Pull request by david-yz-liu**

### CODE REVIEW

File: data/dev/repos/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# markus runtime data, ignore everything except .gitignore
```
Comment (david-yz-liu) [inline data/dev/repos/.gitignore]: @SoftwareDev Does this file need to be removed for something on the git side to work?

File: lib/repo/git_repository.rb [hunk 2]
```diff
@@ -55,15 +64,72 @@ def initialize(connect_string)
...
+      cloned_repo.add(all: true)
+
+      cloned_repo.commit_all('Initial commit.')
+      cloned_repo.push
+
+      # Finished cloning repo
+
+      repo = Rugged::Repository.discover(REPOSITORY_STORAGE + '/' + repo_name)
 
       # Do an initial commit with a README to create index.
```
Comment (david-yz-liu) [inline lib/repo/git_repository.rb]: Is this whole part still necessary? Seems like a repeat of what just happened.

Comment (SoftwareDev) [inline lib/repo/git_repository.rb]: @david-yz-liu Hey David, it looks messy, but what's happening in those lines is that the master branch is being created after the repo itself is created. For some reason Gitolite doesn't generate a master branch when you first create the repo and I found that I needed to this to make it work properly.

File: data/dev/repos/.gitignore (hunk 1 above)
Comment (SoftwareDev) [inline data/dev/repos/.gitignore]: @david-yz-liu I've never seen this file before so I'd get rid of it.

File: lib/repo/git_repository.rb (hunk 2 above)
Comment (david-yz-liu) [inline lib/repo/git_repository.rb]: Hmm okay, thanks @SoftwareDev! @reidka have been trying out the code, and it seems to more or less work. Very exciting stuff!

**Comment by david-yz-liu** [comment]
@SoftwareDev Merged!

**Pull request merged**